### PR TITLE
doc: add volume cloning doc

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind doc

**What this PR does / why we need it**:

doc: add volume cloning doc

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
pls also add a link to https://github.com/kubernetes-sigs/azurefile-csi-driver#examples, follow azure disk example

File: deploy/example/cloning/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Volume Cloning Example
```
Comment (andyzhangx) [inline deploy/example/cloning/README.md:1]: also add this folder to https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/c59a0b75d2094e189fee72b367d543e7a9636674/hack/verify-yamllint.sh#L24

Comment (andyzhangx) [inline deploy/example/cloning/README.md:1]: add 
```
## Feature Status: Beta
```

Comment (umagnus) [reply deploy/example/cloning/README.md:1]: done

Comment (umagnus) [reply deploy/example/cloning/README.md:1]: done

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**